include/xsimd/arch/generic/xsimd_generic_math.hpp

```diff
@@ -1707,6 +1707,40 @@ namespace xsimd {
     }
 
 
+#if !defined(__FAST_MATH__)
```

Comment:
Interesting approach! I wonder if the following works as expected and is faster or not:
```
namespace details {
template<class T>
T force_add(T x, T y) {
#if defined(__FAST_MATH__) && (defined(__clang__) || defined(__GNUC__))
__attribute__((noinline))
#endif
    return x + y;
}
}
```

then call it appropriately

Reply:
Now on $DAILY_JOB.

Without trying I'd say that a competent compiler will still optimize it away.

Reply:
But notice that the optimization we want to break is ffast-math related. Inlining should make no difference.